identity/migration/src/main/java/io/camunda/identity/migration/AuthorizationMigrationHandler.java

```diff
@@ -0,0 +1,134 @@
+                      permissionAndResources.entrySet().stream()
+                          .flatMap(
+                              e ->
+                                  convertPermission(e).stream()
+                                      .map(entry -> new SimpleEntry<>(entry, e.getValue())))
+                          .collect(
+                              Collectors.groupingBy(
+                                  SimpleEntry::getKey,
+                                  Collectors.flatMapping(
+                                      e -> e.getValue().stream(), Collectors.toList())));
+                  final long ownerKey = getOwnerKeyForUsername(owner);
+                  authorizationService.patchAuthorization(
+                      new PatchAuthorizationRequest(
+                          ownerKey,
+                          PermissionAction.ADD,
+                          convertResourceType(resourceType),
+                          permissions));
+
+                  final Collection<UserResourceAuthorization> migrated =
+                      permissionAndResources.entrySet().stream()
+                          .flatMap(
+                              entry ->
+                                  entry.getValue().stream()
+                                      .map(
+                                          resourceId ->
+                                              new UserResourceAuthorization(
+                                                  owner, resourceId, resourceType, entry.getKey())))
+                          .toList();
+                  managementIdentityProxy.markAsMigrated(migrated);
+                });
+          });
+    }
+  }
+
+  private static AuthorizationResourceType convertResourceType(final String resourceType) {
+    if ("process-definition".equalsIgnoreCase(resourceType)) {
+      return AuthorizationResourceType.PROCESS_DEFINITION;
+    }
+    if ("decision-definition".equalsIgnoreCase(resourceType)) {
+      return AuthorizationResourceType.DECISION_DEFINITION;
+    }
+    throw new IllegalArgumentException("Unsupported resource type: " + resourceType);
+  }
+
+  private static List<PermissionType> convertPermission(final Entry<String, List<String>> e) {
+    if (e.getKey().startsWith("read")) {
+      return List.of(PermissionType.READ);
+    }
+    if (e.getKey().startsWith("write")) {
+      return List.of(PermissionType.CREATE, PermissionType.UPDATE, PermissionType.DELETE);
+    }
+
+    if (e.getKey().startsWith("create")) {
+      return List.of(PermissionType.CREATE);
+    }
+
+    if (e.getKey().startsWith("update")) {
+      return List.of(PermissionType.UPDATE);
+    }
+
+    if (e.getKey().startsWith("delete")) {
+      return List.of(PermissionType.DELETE);
+    }
+
+    throw new IllegalArgumentException("Unsupported permission type: " + e.getKey());
+  }
+
+  private long getOwnerKeyForUsername(final String owner) {
+    return 0;
```

Comment:
💭  I think we should leave a todo here so we don't leave the default value in place :)